app/main/views/search.py

```diff
@@ -76,6 +78,10 @@ def supplier_search():
     else:
         roles, raw_role_data = get_all_roles(data_api_client)
 
+    role_filters = {}
+    for role in roles:
+        role_filters[role] = role in selected_roles
+
```

Comment:
you can use dictionary comprehensions for this sort of thing, eg:

`role_filters = {role: role in selected_roles for role in roles}`